rego/rego_test.go

```diff
@@ -1735,6 +1735,38 @@ func TestPrepareWithWasmTargetNotSupported(t *testing.T) {
 	})
 }
 
+func TestEvalWithContextTimeout(t *testing.T) {
+	for _, tgt := range []string{"rego", "wasm"} {
+		t.Run(tgt, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
+			defer cancel()
+
+			// This takes long.
+			policy := `package p
+			allow {
+				net.cidr_expand("1.0.0.0/1")
+			}
+			`
+			pq, err := New(
+				Query("data.p.allow"),
+				Module("p.rego", policy),
+				Target(tgt),
+			).PrepareForEval(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			_, err = pq.Eval(ctx)
+			if err == nil {
+				t.Fatal("expected err, got nit")
+			}
+			if !topdown.IsCancel(err) {
```

Comment:
⚠️ For the wasm target, this won't be true either -- it'll return a different error. But it should _never_ run until completion and return `nil`.